prov/util/src/util_eq.c

```diff
@@ -64,7 +64,7 @@ static ssize_t util_eq_read(struct fid_eq *eq_fid, uint32_t *event,
 	if (event)
 		*event = entry->event;
 	if (buf) {
-		ret = MIN(len, entry->size);
+		ret = MIN(len, (size_t)entry->size);
```

Comment:
```entry``` element has type util_event which is widely used in a lot of providers & changing it's size may affect to stability & performance. I just have no environment to test such changes deeply